Objects/exceptions.c

```diff
@@ -2316,6 +2319,12 @@ MemoryError_dealloc(PyBaseExceptionObject *self)
     _PyObject_GC_UNTRACK(self);
     BaseException_clear(self);
 
+    /* If this is a subclass of MemoryError, we don't need to
+     * do anything in the free-list*/
+    if (Py_TYPE(self) != (PyTypeObject *) PyExc_MemoryError) {
```

Comment:
How about use `!Py_IS_TYPE(self, (PyTypeObject *) PyExc_MemoryError)` in here?

Reply:
+1 with @shihai1991 's comment

Reply:
Done in the last ammend